x-pack/plugins/security_solution/public/cases/components/case_view/index.tsx

```diff
@@ -98,6 +115,39 @@ export const CaseComponent = React.memo<CaseProps>(
       caseId,
     });
 
+    const alertsQuery = useMemo(() => buildAlertsQuery(getRuleIdsFromComments(caseData.comments)), [
+      caseData.comments,
+    ]);
+
+    /**
+     * For the future developer: useSourcererScope is security solution dependent.
+     * You can use useSignalIndex as an alternative.
+     */
+    const { browserFields, docValueFields, selectedPatterns } = useSourcererScope(
+      SourcererScopeName.detections
+    );
+
+    const { loading: isLoadingAlerts, data: alertsData } = useQueryAlerts<
+      BaseHit<SignalHit>,
+      unknown
+    >(alertsQuery, selectedPatterns[0]);
+
+    const alerts = useMemo(
```

Comment:
I think this can belong to `useQueryAlerts`. this hood should be able to chew/format the data the way we want.